File: datagym-backend/src/main/java/ai/datagym/application/labelConfiguration/entity/geometry/LcEntryRectangle.java
package ai.datagym.application.labelConfiguration.entity.geometry;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import java.io.Serializable;

@Entity
@DiscriminatorValue(value = "rectangle")
public class LcEntryRectangle extends LcEntryGeometry implements Serializable {

    public LcEntryRectangle() {
        super();
    }
}

File: datagym-frontend/src/app/label-config/model/geometry/LcEntryRectangle.ts
import { LcEntryGeometry } from './LcEntryGeometry';
import { LcEntryType } from '../LcEntryType';

export class LcEntryRectangle extends LcEntryGeometry {
  static type: LcEntryType = LcEntryType.RECTANGLE;

  constructor(lcEntryParentId: string | null, key?: string, value?: string, color?: string, shortcut?: string) {
    super(LcEntryRectangle.type, lcEntryParentId, key, value, color, shortcut);
  }
}
